### PR TITLE
CDAP-13558 Support CDC pipelines on Spark version 2.x

### PR DESCRIPTION
**Pull request by mlozbin-cybervisiontech**
JIRA - https://issues.cask.co/browse/CDAP-13558
WIKI - https://wiki.cask.co/display/CE/CDC+Solution+Enhancements

In scope of this PR:
- introduced single message format
- added schema validation
- added config validation
- changed exception handling to follow recent guidelines

### CODE REVIEW

**Comment by mlozbin-cybervisiontech** [comment]
@albertshau, could you please take a look.

File: src/main/java/co/cask/cdc/plugins/common/Schemes.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import co.cask.cdap.api.data.schema.Schema.Field;
+import co.cask.cdap.api.data.schema.Schema.Type;
+
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+/**
+ * Helper class with common cdc schemes definitions.
+ */
+public class Schemes {
```
Comment (albertshau) [inline src/main/java/co/cask/cdc/plugins/common/Schemes.java]: nit: Schemas

File: src/main/java/co/cask/cdc/plugins/sink/CDCBigTableConfig.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+  public void validate() {
+    IdUtils.validateId(referenceName);
+    if (!containsMacro(PROJECT) && resolveProject() == null) {
+      throw new InvalidConfigPropertyException("Could not detect Google Cloud project id from the environment. " +
+                                                 "Please specify a project id.", PROJECT);
+    }
+    String serviceAccountFilePath = resolveServiceAccountFilePath();
+    if (!containsMacro(SERVICE_ACCOUNT_FILE_PATH)) {
+      if (serviceAccountFilePath != null) {
```
Comment (albertshau) [inline src/main/java/co/cask/cdc/plugins/sink/CDCBigTableConfig.java]: this can be combined with the previous line to remove one layer of nesting.

**Review by albertshau — COMMENTED** [review]
minor comments

File: src/main/java/co/cask/cdc/plugins/common/Schemes.java (hunk 1 above)
Comment (mlozbin-cybervisiontech) [reply src/main/java/co/cask/cdc/plugins/common/Schemes.java]: fixed

File: src/main/java/co/cask/cdc/plugins/sink/CDCBigTableConfig.java (hunk 2 above)
Comment (mlozbin-cybervisiontech) [reply src/main/java/co/cask/cdc/plugins/sink/CDCBigTableConfig.java]: fixed

**Pull request merged**